pkg/actors/targets/workflow/workflow.go

```diff
@@ -344,15 +341,19 @@ func (w *workflow) createIfCompleted(ctx context.Context, runtimeState *backend.
 }
 
 func (w *workflow) scheduleWorkflowStart(ctx context.Context, startEvent *backend.HistoryEvent, state *wfenginestate.State) error {
+	state.AddToInbox(startEvent)
+	if err := w.saveInternalState(ctx, state); err != nil {
+		return err
+	}
+
 	// Schedule a reminder to execute immediately after this operation. The reminder will trigger the actual
 	// workflow execution. This is preferable to using the current thread so that we don't block the client
 	// while the workflow logic is running.
 	if _, err := w.createReliableReminder(ctx, "start", nil, 0); err != nil {
 		return err
 	}
 
-	state.AddToInbox(startEvent)
-	return w.saveInternalState(ctx, state)
+	return nil
```

Comment:
Which operations are those?

Reply:
```
1. w.saveInternalState
2. w.createReliableReminder
3. w.saveInternalState
```
It might be better to form it as a question: if any of those actions in this sequence fail, do the actions made by previous ones leave any "unfinished" state/reminders that need to be cleaned/reverted?

Reply:
1. Is in-memory AFAIR

2 & 3 are different APIs so cannot be made transactionally. The order they are done ensures we at least 
cannot be put into a broken state AFAIR.

Reply:
agree with Josh, I wanted to have this order of operations for a long time